analytic_engine/src/meta/details.rs

```diff
@@ -291,86 +327,231 @@ impl Manifest for ManifestImpl {
+    const CURRENT_SNAPSHOT_NAME: &str = "current";
+    const SNAPSHOT_PATH_PREFIX: &str = "manifest/snapshot";
+
+    fn current_snapshot_path(&self) -> Path {
+        format!(
+            "{}/{}/{}/{}",
+            Self::SNAPSHOT_PATH_PREFIX,
+            self.space_id,
+            self.table_id,
+            Self::CURRENT_SNAPSHOT_NAME
+        )
+        .into()
+    }
+
+    fn snapshot_path(&self, seq: SequenceNumber) -> Path {
+        format!(
+            "{}/{}/{}/{}",
+            Self::SNAPSHOT_PATH_PREFIX,
+            self.space_id,
+            self.table_id,
+            seq,
+        )
+        .into()
+    }
+
+    async fn load_current_snapshot_seq(&self) -> Result<Option<SequenceNumber>> {
+        let path = self.current_snapshot_path();
+        let payload = {
+            let get_res = self.store.get(&path).await;
+            if let Err(object_store::ObjectStoreError::NotFound { path, source }) = &get_res {
+                warn!(
+                    "Current snapshot file doesn't exist, path:{}, err:{}",
+                    path, source
+                );
+                return Ok(None);
+            }
+
+            get_res
+                .context(FetchCurrentSnapshot)?
+                .bytes()
+                .await
+                .context(FetchCurrentSnapshot)?
+        };
+        let current_snapshot = meta_update::CurrentSnapshot::decode(payload.as_bytes())
+            .context(DecodeCurrentSnapshot)?;
+        Ok(Some(current_snapshot.sequence))
+    }
+}
+
+#[async_trait]
+impl MetaUpdateSnapshotStore for ObjectStoreBasedSnapshotStore {
+    /// Store the latest snapshot to the underlying store, and then update the
+    /// `current_snapshot` to ensure the mapping to the latest snapshot.
+    /// The old snapshot will be purged when things done.
+    async fn store(&self, snapshot: &Snapshot) -> Result<()> {
+        let snapshot_pb = meta_update::Snapshot::from(snapshot.clone());
+        let payload = snapshot_pb.encode_to_vec();
+        let path = self.snapshot_path(snapshot.end_seq);
+        self.store
+            .put(&path, payload.into())
+            .await
+            .context(StoreSnapshot)?;
+
+        // Fetch the old snapshot sequence before overwriting it.
+        let old_snapshot_seq = match self.load_current_snapshot_seq().await {
+            Ok(v) => v,
+            Err(e) => {
+                warn!(
+                    "Ignore the failure to load current snapshot for following purge, err:{}",
+                    e
+                );
+                None
+            }
+        };
+
+        // Update the current snapshot.
+        let current_snapshot = meta_update::CurrentSnapshot {
+            sequence: snapshot.end_seq,
+        };
+        let payload = current_snapshot.encode_to_vec();
+        let path = self.current_snapshot_path();
+        self.store
+            .put(&path, payload.into())
+            .await
+            .context(UpdateCurrentSnapshot)?;
```

Comment:
Maybe add comment to decribe the corruption risk?

Reply:
Done.